### PR TITLE
Removing restrictions should not show "Removed Public restriction"

### PR DESCRIPTION
**Pull request by hazho**
currently when the page editor removes the restrictions, the page and site history shows like (Removed the 'Public' view restriction), and this off course is not making any sense, this PR is just simply removing the variables and shows that the restrictions are removed

### CODE REVIEW

**Comment by zerolab** [comment]
Hey @hazho 

This is a good catch. it looks like the restriction type set on deletion is `none` hence the default. We can do this, in which case the `PageViewRestriction.delete()` log action would need updating to remove the extra context. A better approach would be to provide the correct restriction type and tweak the message. Something like:

<details>
<summary>Suggested diff</summary>

```diff
diff --git a/wagtail/core/models/__init__.py b/wagtail/core/models/__init__.py
index 5b14a7f5d5..6f02feb3cf 100644
--- a/wagtail/core/models/__init__.py
+++ b/wagtail/core/models/__init__.py
@@ -3127,10 +3127,14 @@ class PageViewRestriction(BaseViewRestriction):
         """
         Custom delete handler to aid in logging
         :param user: the user removing the view restriction
-        :param specific_instance: the specific model instance the restriction applies to
         """
         specific_instance = self.page.specific
         if specific_instance:
+            removed_restriction_type = (
+                PageViewRestriction.objects.filter(page=self.page)
+                .values_list('restriction_type', flat=True)[0]
+            )
+
             log(
                 instance=specific_instance,
                 action='wagtail.view_restriction.delete',
@@ -3138,7 +3142,7 @@ class PageViewRestriction(BaseViewRestriction):
                 data={
                     'restriction': {
                         'type': self.restriction_type,
-                        'title': force_str(dict(self.RESTRICTION_CHOICES).get(self.restriction_type))
+                        'title': force_str(dict(self.RESTRICTION_CHOICES).get(removed_restriction_type))
                     }
                 }
             )
diff --git a/wagtail/core/wagtail_hooks.py b/wagtail/core/wagtail_hooks.py
index 9dace5141a..044adf045b 100644
--- a/wagtail/core/wagtail_hooks.py
+++ b/wagtail/core/wagtail_hooks.py
@@ -284,7 +284,7 @@ def register_core_log_actions(actions):
 
         def format_message(self, log_entry):
             try:
-                return _("Removed the '%(restriction)s' view restriction") % {
+                return _("Removed the '%(restriction)s' view restriction. The page is now public.") % {
                     'restriction': log_entry.data['restriction']['title'],
                 }
             except KeyError:
```
</details>

Plus https://github.com/wagtail/wagtail/blob/main/wagtail/admin/tests/test_audit_log.py#L94 would need updating

**Comment by hazho** [comment]
> Hey @hazho
> 
> This is a good catch. it looks like the restriction type set on deletion is `none` hence the default. We can do this, in which case the `PageViewRestriction.delete()` log action would need updating to remove the extra context. A better approach would be to provide the correct restriction type and tweak the message, so as to continue having the context. Something like:
> 
> Suggested diff
> Plus https://github.com/wagtail/wagtail/blob/main/wagtail/admin/tests/test_audit_log.py#L94 would need updating

I'll leave this for you (https://github.com/wagtail/wagtail/blob/main/wagtail/admin/tests/test_audit_log.py#L94)

**Comment by hazho** [comment]
I'm not so good at testing framework

**Comment by hazho** [comment]
sorry guys, I didn't test the changes earlier, now I did, but it is not fetching the removed restriction_type, need to work on it deeper(,I think meanwhile the first suggestion is viable(just removing the restriction_type title from the message), after I or any of you work on this deeper, then we can change it by new PR..!

**Comment by hazho** [comment]
OK, currently the PageViewRestriction has been updated to fulfill the needs of this PR

**Comment by hazho** [comment]
I have another PR, shall I wait till you merge this one, or merge both PRs? it is about the issue #7483 ( https://github.com/wagtail/wagtail/pull/7567/commits/fd2d90a6c069f664c928ec954491256e82bbcaec )

File: wagtail/core/wagtail_hooks.py [hunk 1]
```diff
@@ -284,9 +284,10 @@ class DeleteViewRestrictionActionFormatter(LogFormatter):
 
         def format_message(self, log_entry):
             try:
-                return _("Removed the '%(restriction)s' view restriction") % {
+                return _("Removed the '%(restriction)s' view restriction. The page is public now.") % {
```
Comment (zerolab) [inline wagtail/core/wagtail_hooks.py:287]:
```suggestion
                return _("Removed the '%(restriction)s' view restriction. The page is public.") % {
```

let's avoid confusion

**Comment by hazho** [comment]
> @hazho as for #7483, please keep separate issue fixes in separate PRs

can I create a separate PR while this is not yet merged? I think it can't be, github is mixing the commits together

**Comment by hazho** [comment]
Guys, I have many other PRs, waiting to know the way of creating a separate PR for new commits(if applicable) or please merge this (if there is no issues with it)

**Comment by lb-** [comment]
@hazho just checking in if you are still ok to resolve the feedback above?

You will also need to rebase these changes on the latest wagtail main as there are conflicts.

**Comment by hazho** [comment]
I tryed this, but can't resolve the conflicts, not by CLI nor manully, Iwill try to create another fork and re-pull this request again

**Comment by hazho** [comment]
image
can't resolve manually, nore automatically, I have to create a new repo, and make changes, but, ZeroLabs's contributions will be erased(sorry for that).. if that is fine, then I will close this PR

**Comment by lb-** [comment]
>  can't resolve manually, nore automatically, I have to create a new repo, and make changes, but, ZeroLabs's contributions will be erased(sorry for that).. if that is fine, then I will close this PR

Hey @hazho did you want to reach out on Slack? I can help you create a branch with these changes. I personally use [Fork](https://git-fork.com/) as I find it easier than the command line. 

Roughly this is what you need to do
- pull your remote Wagtail branch locally (E.g. it might be upstream/...)
- check out the upstream/main branch as something like main-backup
- create a new branch called fix/restrictions-messages
- now, cherry pick each individual commit, from the oldest to newest that relates to this PR onto that branch, resolve conflicts as they are needed
- then check out the main branch and reset (hard) to the upstream/main branch, you will need to force push this to your remote repo
- now checkout your fix branch and do a rebase with a squash on all of your commits (so there is one commit)
- push this fix branch to your remote, load up GitHub Wagtail and you will see a message asking if you want to create a PR

**Comment by hazho** [comment]
I will try with gitlab mirror, if things are going to be flexible, then I will contrib, otherwise I quite....!
Wht github is so complicating things, I used gitlab less than github, but I loved the first one because the second one is really sucks..!

**Comment by lb-** [comment]
Ok no problems, I think the issue can be resolved on your local machine much easier and the main problem is that you have made changes on your clone's main branch instead of creating a new branch with the changes. 

But if gitlab works for you, go for it. Reach out if you need a hand, even happy to line up a screenshare at some point if you need.

**Comment by hazho** [comment]
> Ok no problems, I think the issue can be resolved on your local machine much easier and the main problem is that you have made changes on your clone's main branch instead of creating a new branch with the changes.
> 
> But if gitlab works for you, go for it. Reach out if you need a hand, even happy to line up a screenshare at some point if you need.

the issue is not the lack of the knowledge, but the unneccesary steps

**Comment by hazho** [comment]
> > can't resolve manually, nore automatically, I have to create a new repo, and make changes, but, ZeroLabs's contributions will be erased(sorry for that).. if that is fine, then I will close this PR
> 
> Hey @hazho did you want to reach out on Slack? I can help you create a branch with these changes. I personally use [Fork](https://git-fork.com/) as I find it easier than the command line.
> 
> Roughly this is what you need to do
> 
> * pull your remote Wagtail branch locally (E.g. it might be upstream/...)
> * check out the upstream/main branch as something like main-backup
> * create a new branch called fix/restrictions-messages
> * now, cherry pick each individual commit, from the oldest to newest that relates to this PR onto that branch, resolve conflicts as they are needed
> * then check out the main branch and reset (hard) to the upstream/main branch, you will need to force push this to your remote repo
> * now checkout your fix branch and do a rebase with a squash on all of your commits (so there is one commit)
> * push this fix branch to your remote, load up GitHub Wagtail and you will see a message asking if you want to create a PR

@lb- here I have tried Gitlab, but was One way, I couldn't merge the request(PR) from gitlab to github, even with my own repositories..!
anyway, I your steps above, but the same issue..!
I also tried to create a second repository of Live updates of wagtail/wagtail, called hazho/wagtail, but the PR can't be done to the original one, instead, it can be done from the branch to the main of hazho/wagtail..!
I think I have to delete hazho/wagtailo (which has 4 PRs) I don't wanna delete all of them, I hope I could just fetch the new updates from wagtail/wagtail to hazho/wagtailo, and ignore this PR, but the fetching is not possible due to large amount of commits since last time I fetched..!

**Comment by rohitsrma** [comment]
Can i work on this?

**Comment by lb-** [comment]
@rohitsrma go for it, from what I can see - the problem is something we do want to see fixed. Please do review all the comments and feedback thoroughly though. Thank you.

**Comment by lb-** [comment]
Closing in favour of a new PR that replaces this. https://github.com/wagtail/wagtail/pull/11044

**Pull request closed without merging**